packages/relay-runtime/store/__tests__/observeFragment-test.js

```diff
@@ -337,6 +337,189 @@ test('read deferred fragment', async () => {
+      nodes(ids: ["1", "2"]) {
+        ...observeFragmentTestPluralThrowOnFieldErrorFragment
+      }
+    }
+  `;
+
+  const fragment = graphql`
+    fragment observeFragmentTestPluralThrowOnFieldErrorFragment on User
+    @relay(plural: true)
+    @throwOnFieldError {
+      name
+    }
+  `;
+
+  let dataSource: Sink<GraphQLResponse>;
+  const fetch = (
+    _query: RequestParameters,
+    _variables: Variables,
+    _cacheConfig: CacheConfig,
+  ) => {
+    // $FlowFixMe[missing-local-annot] Error found while enabling LTI on this file
+    return RelayObservable.create(sink => {
+      dataSource = sink;
+    });
+  };
+
+  const environment = createMockEnvironment({
+    network: RelayNetwork.create(fetch),
+  });
+  const variables = {};
+  const operation = createOperationDescriptor(query, variables);
+
+  environment.execute({operation}).subscribe({});
+  invariant(dataSource != null, 'Expected data source to be set');
+  dataSource.next({
+    data: {
+      nodes: [
+        {id: '1', __typename: 'User', name: null},
+        {id: '2', __typename: 'User', name: 'Bob'},
+      ],
+    },
+    errors: [{message: 'error', path: ['nodes', 0, 'name']}],
+  });
+
+  const {data} = environment.lookup(operation.fragment);
+  // $FlowFixMe Data is untyped
+  const observable = observeFragment(environment, fragment, data.nodes);
+  withObservableValues(observable, results => {
+    expect(results).toEqual([
+      {
+        error: new Error(
+          'Relay: Unexpected response payload - check server logs for details.',
+        ),
+        state: 'error',
+      },
+    ]);
+  });
+});
+
+test('Resolver error with @relay(plural: true) @throwOnFieldError', async () => {
+  const query = graphql`
+    query observeFragmentTestResolverErrorWithPluralThrowOnFieldErrorQuery {
+      nodes(ids: ["7", "8"]) {
+        ...observeFragmentTestResolverErrorWithPluralThrowOnFieldErrorFragment
+      }
+    }
+  `;
+
+  const fragment = graphql`
+    fragment observeFragmentTestResolverErrorWithPluralThrowOnFieldErrorFragment on User
+    @relay(plural: true)
+    @throwOnFieldError {
+      always_throws
+    }
+  `;
+
+  const environment = createMockEnvironment({
+    store: new LiveResolverStore(new RelayRecordSource()),
+  });
+  const variables = {};
+  const operation = createOperationDescriptor(query, variables);
+  environment.commitPayload(operation, {
+    nodes: [
+      {id: '7', __typename: 'User'},
+      {id: '8', __typename: 'User'},
+    ],
+  });
+  const {data} = environment.lookup(operation.fragment);
+  // $FlowFixMe Data is untyped
+  const observable = observeFragment(environment, fragment, data.nodes);
+  withObservableValues(observable, results => {
+    expect(results).toEqual([
+      {
+        error: new Error(
+          "Relay: Resolver error at path 'always_throws' in 'observeFragmentTestResolverErrorWithPluralThrowOnFieldErrorFragment'.",
+        ),
+        state: 'error',
+      },
+    ]);
+  });
+});
+
```

Comment:
Can we add some additional tests:

1. A test demonstrating a store update/publish that touches multiple items in the list triggering multiple observable updates?
2. A test or two demonstrating that `waitForFragmentData` also works with plural fragments?

Reply:
https://github.com/facebook/relay/pull/4862/commits/e596faac720d127baea60f84ee2c9b286ffbc72d